contracts/src/bin/vendor.rs

```diff
@@ -65,38 +22,122 @@ fn main() {
 }
 
 fn run() -> Result<()> {
-    let artifacts = paths::contract_artifacts_dir();
-    fs::create_dir_all(&artifacts)?;
+    let vendor = Vendor::new()?;
 
-    log::info!("vendoring contract artifacts to '{}'", artifacts.display());
-    for (npm_path, local_path) in NPM_CONTRACTS {
-        log::info!("retrieving {}", npm_path);
-        let source = Source::npm(npm_path.to_string());
+    vendor
+        .full()
+        .github(
+            "BalancerV2Vault",
+            "balancer-labs/balancer-core-v2/v1.0.0-artifacts/\
+             deployments/artifacts/mainnet/Vault.json",
+        )?
+        .npm(
+            "ERC20Mintable",
+            "@openzeppelin/contracts@2.5.0/build/contracts/ERC20Mintable.json",
+        )?
+        .npm(
+            "GPv2AllowListAuthentication",
+            // We use `_Implementation` because the use of a proxy contract makes
+            // deploying  for the e2e tests more combersome.
+            "@gnosis.pm/gp-v2-contracts@0.0.1-alpha.15/\
+             deployments/mainnet/GPv2AllowListAuthentication_Implementation.json",
+        )?
+        .npm(
+            "GPv2Settlement",
+            "@gnosis.pm/gp-v2-contracts@0.0.1-alpha.15/deployments/mainnet/GPv2Settlement.json",
+        )?
+        .npm(
+            "UniswapV2Factory",
+            "@uniswap/v2-core@1.0.1/build/UniswapV2Factory.json",
+        )?
+        .npm(
+            "UniswapV2Router02",
+            "@uniswap/v2-periphery@1.1.0-beta.0/build/UniswapV2Router02.json",
+        )?
+        .npm("WETH9", "canonical-weth@1.4.0/build/contracts/WETH9.json")?;
+
+    vendor
+        .abi_only()
+        .npm(
+            "ERC20",
+            "@openzeppelin/contracts@3.3.0/build/contracts/ERC20.json",
+        )?
+        .npm(
+            "IUniswapLikeRouter",
+            "@uniswap/v2-periphery@1.1.0-beta.0/build/IUniswapV2Router02.json",
+        )?
+        .npm(
+            "IUniswapLikePair",
+            "@uniswap/v2-periphery@1.1.0-beta.0/build/IUniswapV2Pair.json",
+        )?;
+
+    Ok(())
+}
+
+struct Vendor {
+    artifacts: PathBuf,
+}
+
+impl Vendor {
+    fn new() -> Result<Self> {
+        let artifacts = paths::contract_artifacts_dir();
+        log::info!("vendoring contract artifacts to '{}'", artifacts.display());
+        fs::create_dir_all(&artifacts)?;
+        Ok(Self { artifacts })
+    }
+
+    fn full(&self) -> VendorContext {
```

Comment:
maybe add some comments here that full is needed if you want to deploy the contract (e.g. for e2e tests)